### PR TITLE
Feature: shopping cart

### PR DESCRIPTION
**Pull request by RichardIrala**
Files changed:
* atoms/Item/Item.jsx
* functions/web2/web2.js
* functions/web3/web3.js
* pages/inventory.js
* web3-abis/marketplaceAbi.js

Feature ready. It shouldn't be merged until everything is synchronized and deployed on the xerial API

### CODE REVIEW

File: atoms/Item/Item.jsx [hunk 1]
```diff
@@ -20,10 +20,10 @@ const Item = ({ nft, sellerAddress, XerialWalletViewmodel }) => {
             collectionAddress: nft.metadata.contract.address,
           });
         } else {
-          await web3Functions.purchaseNfts({
+          await web3Functions.purchaseNft({
             tokenTypeId: nft.typeId,
             collectionAddress: nft.metadata.contract.address,
-            amount: amount,
+            quantity: amount,
```
Comment (luislucena16) [inline atoms/Item/Item.jsx:26]: great! this was the change we talked about, `quantity` is more clear with the function 👍

File: functions/web3/web3.js [hunk 2]
```diff
@@ -25,13 +25,48 @@ async function connectToMetaMask() {
...
+    for (const nft of nfts) {
+      const { tokenTypeId, quantity, collectionAddress } = nft;
+      if (!tokenTypeId || !quantity || !collectionAddress) {
+        throw new Error("Each object in the array must have the properties tokenTypeId, quantity and collectionAddress");
+      }
+    }
+
+    const usdcAddress = process.env.NEXT_PUBLIC_POLYGON_USDC_CONTRACT;
+    // const marketplaceAddress = process.env.NEXT_PUBLIC_POLYGON_MARKETPLACE_CONTRACT;
+    const marketplaceAddress = "0x77533EA81cd5CAAeE87E1D489261B4B1740DfAd2";
```
Comment (luislucena16) [inline functions/web3/web3.js]: comment on this and uncomment the above line, no hardcode values, set it in `.env`

File: pages/inventory.js [hunk 3]
```diff
@@ -84,12 +84,16 @@ const Inventory = () => {
...
-    setPlayerItemsOnSecondaryMarket(items);
+    try {
+      const items = await web2Functions.getPlayerItemsOnSecondaryMarket({
+        chain: "polygon",
+        userAddress: userAddress,
+        studioAddress: project.address,
+      });
+      setPlayerItemsOnSecondaryMarket(items);
+    } catch (error) {
+      console.error("error to load playerItemsOnSecondaryMarket")
```
Comment (luislucena16) [inline pages/inventory.js]: rename word `error` to `Error`

**Review by luislucena16 — CHANGES_REQUESTED** [review]

File: functions/web3/web3.js (hunk 2 above)
Comment (RichardIrala) [reply functions/web3/web3.js]: Done

File: pages/inventory.js (hunk 3 above)
Comment (RichardIrala) [reply pages/inventory.js]: Done

**Comment by luislucena16** [comment]
all good now! this PR is not yet finalised! we will still be checking

File: functions/web3/web3.js [hunk 4]
```diff
@@ -79,7 +79,7 @@ web3Functions.purchaseNft = async function ({ tokenTypeId, quantity, collectionA
       const resTx = await tx.wait();
     }
 
-    const purchaseTransaction = await marketplace.connect(signer).primaryPurchase(nft);
+    const purchaseTransaction = await marketplace.connect(signer).primaryPurchase([{ tokenTypeId, quantity, collectionAddress }]);
```
Comment (luislucena16) [inline functions/web3/web3.js:84]: I don't see this! good job!

File: functions/web3/web3.js [hunk 5]
```diff
@@ -25,13 +25,47 @@ async function connectToMetaMask() {
 
 const web3Functions = {};
 
-web3Functions.purchaseNfts = async function ({ tokenTypeId, amount, collectionAddress }) {
+web3Functions.purchaseNfts = async function (nfts) {
```
Comment (luislucena16) [inline functions/web3/web3.js:29]: add a descriptive comment in this function

File: functions/web3/web3.js [hunk 6]
```diff
@@ -25,13 +25,47 @@ async function connectToMetaMask() {
...
+
+    const purchaseTransaction = await marketplace.connect(signer).primaryPurchase(nfts);
+    const purchaseTransactionWaited = await purchaseTransaction.wait();
+    console.log(purchaseTransactionWaited);
+  } catch (error) {
+    throw new Error(error.message);
+  }
+};
+
+web3Functions.purchaseNft = async function ({ tokenTypeId, quantity, collectionAddress }) {
```
Comment (luislucena16) [inline functions/web3/web3.js:64]: add a descriptive comment in this function

**Review by luislucena16 — CHANGES_REQUESTED** [review]

File: functions/web3/web3.js (hunk 5 above)
Comment (RichardIrala) [reply functions/web3/web3.js:29]: Done

File: functions/web3/web3.js (hunk 6 above)
Comment (RichardIrala) [reply functions/web3/web3.js:64]: Done

**Review by luislucena16 — APPROVED** [review]

**Pull request merged**